src/ipc/capnp/protocol.cpp

```diff
@@ -30,10 +30,33 @@
 namespace ipc {
 namespace capnp {
 namespace {
-void IpcLogFn(bool raise, std::string message)
+
+BCLog::Level ConvertIPCLogLevel(mp::Log level)
+{
+    switch (level) {
+        case mp::Log::Trace: return BCLog::Level::Trace;
+        case mp::Log::Debug: return BCLog::Level::Debug;
+        case mp::Log::Info: return BCLog::Level::Info;
+        case mp::Log::Warning: return BCLog::Level::Warning;
+        case mp::Log::Error: return BCLog::Level::Error;
+        case mp::Log::Raise: return BCLog::Level::Error;
+    } // no default case, so the compiler can warn about missing cases
+    assert(false);
```

Comment:
I'm torn about what to do here, so feedback welcome. Adding another value upstream at any point would cause a crash here. But because we don't know what that future value would be, we can't predict how we should react to it now.

Just commenting because this effectively freezes `mp::Log`, and would require adding new fields to `LogMessage` instead. I'm ok with that, but @ryanofsky might not be.

Reply:
In c3f700feb6f650a7ceb62714f2d8eb0d5f1cb2ae _multiprocess: align our logging with libmultiprocess's_:

Adding a level to `mp::Log` could be considered a breaking change, similar to changing a function signature in the `.capnp` file in a breaking way.

But we don't want to treat IPC clients as trusted (in the long run), so crashing doesn't seem right. Maybe instead disconnect and log at the error level?

Reply:
Agreed about avoiding crashing.

I pushed a change that does something different than what you suggested though: assume an unknown log level should only be seen at our most verbose level, which is Trace.

I think that's ok, as it should cause a compilation warning (missed switch value), but otherwise not much change for the user. As it would be an API break that requires downstream attention, I think that's reasonable.